web/src/ui_init.js

```diff
@@ -269,7 +269,11 @@ export function initialize_kitchen_sink_stuff() {
 
     message_viewport.$scroll_container.on("wheel", (e) => {
         const delta = e.originalEvent.deltaY;
-        if (!overlays.is_overlay_or_modal_open() && narrow_state.is_message_feed_visible()) {
+        if (
+            !overlays.is_overlay_or_modal_open() &&
+            !popovers.any_active() &&
+            narrow_state.is_message_feed_visible()
+        ) {
```

Comment:
So the change that introduced this bug is that  `message_viewport.$scroll_container` was changed to be the entire page, and it's possible we want to just register this wheel handler more narrowly instead? I'm not sure.

https://chat.zulip.org/#narrow/stream/9-issues/topic/emoji.20popover.20scrolls.20in.20message.20feed.20as.20well/near/1609664

Reply:
Even if we narrow the scope of this wheel handler, I still think we want to stop changing the selected message if the mouse wheel is scrolling while a popover is open.

First, because there's the bug where the popovers aren't necessarily using the `data-message-id` field and are instead looking at the selected message in the list. So until that's fixed, we definitely don't want this happening.

And second, fidgety mouse wheel scrolling is a thing, and we handle that in the middle of the message feed by closing the popover. But if someone is at the top or bottom of the feed and scrolls while the mouse is out of the popover (by accident or whatever), I don't think we want the blue box / selected message moving around.

Here's a screencast with the scope narrowed to `.app`, so when we scroll when hovering over the emoji popovers (or other popovers) the selected message doesn't change. But we're not stopping the mousewheel handler when a popover is active, so if the mouse hovers else where, then the selected message still moves around. I used a view with only a few messages since it's easy to show how the popover won't close with scrolling because we're at the top and bottom of the feed.

<details>
<summary>Allowing mouse wheel handler when popover is open</summary>

**Note that if this were a long message feed and we were in the middle of the view, then the popovers would all close and the message feed would scroll up or down.**

[Screencast from 2023-07-14 13-52-29.webm](https://github.com/zulip/zulip/assets/63245456/09c772bb-239a-4aa1-83ff-467fee46ee3e)
</details>

---

Note that I don't think we want to limit the scope to the `#message_feed_container` because that makes the scrolling behavior for the blue box / selected message feel really inconsistent with the scrolling behavior of the app. 

Because in the middle of the feed we can scroll many places on the app and the selected message responds, but once you get to the top/bottom, then the selected message will only scroll if the mouse is in the center area.
<details>
<summary>Mouse wheel handler for selected message at top/bottom on #message_feed_container</summary>

[Screencast from 2023-07-14 13-58-11.webm](https://github.com/zulip/zulip/assets/63245456/bc4cbabe-f0db-4635-9b2b-62d7d66dc576)
</details>